agent/doc/swdesign/README.md

```diff
@@ -161,6 +161,459 @@ Needs:
+`swdd~agent-stores-workload-states-of-its-workloads~1`
+
+Status: approved
+
+The AgentManager shall store the workload states of the workloads it manages into the ParameterStorage.
+
+Comment: Empty workload states are not stored and skipped.
+
+Rationale: This ensures that the execution states of a workload and its inter-workload dependencies on the same agent are available.
+
+Tags:
+- AgentManager
+- ParameterStorage
+
+Needs:
+- impl
+- utest
+
+#### AgentManager shall send the workload states of the workload it manages to the server
+`swdd~agent-sends-workload-states-of-its-workloads-to-server~1`
+
+Status: approved
+
+When the AgentManager receives the workload states of the workload it manages, then the AgentManager shall send the workload states to the Ankaios server.
+
+Comment: Empty workload states are not sent.
+
+Rationale: The Ankaios server stores the workload states of the workloads managed by the Ankaios agent.
+
+Tags:
+- AgentManager
+
+Needs:
+- impl
+- utest
+
+#### Agent shall trigger workloads if its inter-workload dependencies are fulfilled
+`swdd~agent-updates-workloads-with-fulfilled-dependencies~1`
+
+Status: approved
+
+When the agent receives an `UpdateWorkloadState` or an `UpdateWorkload` message, then the agent shall create, update or delete workloads having fulfilled inter-workload dependency conditions.
+
+Comment: The Ankaios agent receives workload states from the Ankaios server and from the workloads the agent manages itself. Empty workload states are ignored.
+
+Rationale: Whenever the agent receives new workload states or updates of workloads, the dependencies of a workload might be fulfilled.
+Tags:
+- AgentManager
+- RuntimeManager
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### Agent shall enqueue workload operations with unfulfilled inter-workload dependency states into waiting queue
+`swdd~agent-enqueues-workload-operations-with-unfulfilled-dependencies~1`
+
+Status: approved
+
+When the RuntimeManager receives a `WorkloadOperation` containing a workload with an unfulfilled inter-workload dependency,
+the WorkloadScheduler shall put the workload operation into a waiting queue.
+
+Rationale: The workload shall only be started when its inter-workload dependencies are in the desired configured state.
+
+Tags:
+- RuntimeManager
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### Agent handles UpdateWorkload requests from the server
+`swdd~agent-handles-update-workload-requests~1`
+
+Status: approved
+
+When the AgentManager receives an `UpdateWorkload` message from the server, then the RuntimeManager shall update the workloads the message contains.
+
+Comment: The `UpdateWorkload` message contains workloads to create, update and delete.
+
+Rationale: This ensures separation between receiving the requests from the server and performing the requested actions.
+
+Tags:
+- AgentManager
+- RuntimeManager
+
+Needs:
+- impl
+- utest
```

Comment:
Shall I add stest for this? I think we have already linked stests for triggering workload operations like create, update and deletes. This is just a detail that was missing to link the AgentManager  to the RuntimeManager. This is why I added this req.

Reply:
I don't think that stest is needed, as AgentManager and RuntimeManager are "internal components".